### PR TITLE
LUCENE-8519 MultiDocValues.getNormValues should not call getMergedFieldInfos

### PR DESCRIPTION
**Pull request by shahrs87**
# Description

Please provide a short description of the changes you're making with this pull request.

# Solution

Please provide a short description of the approach taken to implement your solution.

# Tests

Please describe the tests you've developed or run to confirm this patch implements the feature or solves the problem.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://github.com/apache/lucene/blob/main/CONTRIBUTING.md) and my code conforms to the standards described there to the best of my ability.
- [ ] I have given Lucene maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `main` branch.
- [x] I have run `./gradlew check`.
- [ ] I have added tests for my changes.

### CODE REVIEW

**Comment by shahrs87** [comment]
Hi @dsmiley 
Can you please help me review this patch ? I have tried to implement this using your suggestion in the jira. Thank you.

File: lucene/CHANGES.txt [hunk 1]
```diff
@@ -38,6 +38,8 @@ Improvements
 * LUCENE-10416: Update Korean Dictionary to mecab-ko-dic-2.1.1-20180720 for Nori.
   (Uihyun Kim)
 
+* LUCENE-8519 MultiDocValues.getNormValues should not call getMergedFieldInfos (Rushabh Shah)
```
Comment (dsmiley) [inline lucene/CHANGES.txt:41]: But this is an Optimization (should thus go right below); no?

Comment (shahrs87) [reply lucene/CHANGES.txt]: True. Changed it in latest commit. Please review again.

Comment (dsmiley) [reply lucene/CHANGES.txt]: And you've put this in the 10.0 changes but I see no reason not to backport to 9.x.  There's a feature-freeze for 9.2 (it's going to be released) so... we could just wait a week or two here for the 9.3 section to appear by @romseygeek (the RM).

**Review by dsmiley — APPROVED** [review]

File: lucene/CHANGES.txt (hunk 1 above)
Comment (shahrs87) [reply lucene/CHANGES.txt]: I am pretty new to this project. This is my 2nd commit. So I don't know much about the release versions. Just that I understand clearly, we will wait for couple of weeks and once 9.2 is released and 9.3's section is created, I need to change CHANGES.txt and then we will merge this PR ?

Comment (romseygeek) [reply lucene/CHANGES.txt]: I've created the 9.2 branch, so feel free to backport this to 9.x and put it in the 9.3 CHANGES section.

File: lucene/core/src/java/org/apache/lucene/index/MultiDocValues.java [hunk 2]
```diff
@@ -53,8 +53,18 @@ public static NumericDocValues getNormValues(final IndexReader r, final String f
...
+    boolean normFound = false;
+    for (LeafReaderContext leaf : leaves) {
+      LeafReader reader = leaf.reader();
+      FieldInfo info = reader.getFieldInfos().fieldInfo(field);
+      if (info != null && info.hasNorms()) {
+        normFound = true;
+        break;
+      }
+    }
+    if (!normFound) {
```
Comment (mikemccand) [inline lucene/core/src/java/org/apache/lucene/index/MultiDocValues.java:67]: Maybe use `normFound == false` instead?  (For better readability and to reduce the risk of future refactoring bugs).

File: lucene/CHANGES.txt [hunk 3]
```diff
@@ -40,7 +40,7 @@ Improvements
 
 Optimizations
 ---------------------
-(No changes)
+* LUCENE-8519 MultiDocValues.getNormValues should not call getMergedFieldInfos (Rushabh Shah)
```
[the remark this answers is not in the document]
Comment (shahrs87) [reply lucene/CHANGES.txt:43]: Just to understand the process, I will have to create 2 PR's, one for `main` branch and other for `branch_9x`, correct ? @mikemccand  @dsmiley @romseygeek

File: lucene/CHANGES.txt (hunk 1 above)
Comment (shahrs87) [reply lucene/CHANGES.txt]: Just to understand the process, I will have to create 2 PR's, one for main branch and other for branch_9x, correct ?

File: lucene/CHANGES.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (dsmiley) [reply lucene/CHANGES.txt:43]: As a contributor, you can just concern yourself with main.  After merging your PR to main, I'll do a back-port to 9x.  If it's non-trivial, I'll submit a PR.

[the remark this answers is not in the document]
Comment (shahrs87) [reply lucene/CHANGES.txt]: @dsmiley  Hopefully now I got the changes right. Thank you for your patience. :)

**Review by jpountz — APPROVED** [review]

**Comment by dsmiley** [comment]
I ended up cherry picking #902 so I'll close this.

**Pull request closed without merging**